tools/src/dbgen/Lib.hs

```diff
@@ -172,21 +202,85 @@ generateWalletDB CLI{..} spec@GenSpec{..} = do
 
     -- TODO(ks): Simplify this?
     let fakeUtxoSpec = fakeUtxoCoinDistr walletSpec
+    let fakeTxs      = fakeTxsHistory walletSpec
+
     case addTo of
-        Just accId -> case fakeUtxoSpec of
-            NoDistribution -> addAddressesTo accId spec
-            _              -> generateFakeUtxo fakeUtxoSpec accId
+        Just accId ->
+            if (checkIfAddTo fakeUtxoSpec fakeTxs) then
+                addAddressesTo spec accId
+            else do
+                void $ generateFakeUtxo fakeUtxoSpec accId
+                void $ generateFakeTxs fakeTxs accId
+
         Nothing -> do
             say $ printf "Generating %d wallets..." wallets
             wallets' <- timed (forM [1..wallets] genWallet)
             forM_ (zip [1..] wallets') (genAccounts spec)
     say $ green "OK."
+  where
+    checkIfAddTo :: FakeUtxoCoinDistribution -> FakeTxsHistory -> Bool
+    checkIfAddTo NoDistribution NoHistory = True
+    checkIfAddTo _              _         = False
+
+
+-- | Here we generate fake txs. For now it's a simple arbitrary generation.
+generateFakeTxs :: FakeTxsHistory -> AccountId -> UberMonad ()
+generateFakeTxs NoHistory _                = pure ()
+generateFakeTxs SimpleTxsHistory{..} aId   = do
+
+    db <- askWalletDB
+
+    -- Get the number of txs we need to generate.
+    let txsNumber = fromIntegral txsCount
+
+    accounts <- getAccounts Nothing -- aId
+
+    let accountsAddrs :: [Address]
+        accountsAddrs = rights . map unwrapCAddress $ concatMap caAddresses accounts
+
+    let outputAddresses :: [Address]
+        outputAddresses = take txsNumber accountsAddrs
+
+    let genTxOut :: Gen [TxOut]
+        genTxOut = forM outputAddresses $ \address -> TxOut address <$> genCoins
+
+    let genTxHistoryEntry :: IO TxHistoryEntry
+        genTxHistoryEntry = generate $ THEntry
+                                <$> arbitrary
+                                <*> arbitrary -- TODO(ks): Maybe use txOut as well?
+                                <*> arbitrary
+                                <*> genTxOut
+                                <*> pure outputAddresses
+                                <*> arbitrary
+
+    let walletId :: CId Wal
+        walletId = aiWId aId
+
+    -- Generate arbitrary tx ids and txs for the fake history.
+    fakeTxIds <- liftIO $ replicateM txsNumber genTxId
+    fakeTxs   <- liftIO $ replicateM txsNumber genTxHistoryEntry
+
+    let fakeMapTxs :: Map TxId TxHistoryEntry
+        fakeMapTxs = fromList $ zip fakeTxIds fakeTxs
+
+    -- Insert into the @WalletStorage@.
+    insertIntoHistoryCache db walletId fakeMapTxs
+  where
+
+    genTxId :: IO TxId
+    genTxId = generate arbitrary
+
+    genCoins :: Gen Coin
+    genCoins = mkCoin <$> choose (1, 1000)
```

Comment:
How do you ensure that sum of inputs is not less than sum of outputs? This is validated when history is displayed.

Reply:
I'm not sure about this one since it could be an issue, but AFAIK wallet utxo doesn't need to be completely valid - the same principle is used when generating "fake" coins which is done in the previous PR. 
The application is for testing only anyways.

Reply:
I took another look, seems like `getHistory` won't crash because of that after all, false alert